tests/console/generate_dud.pm

```diff
@@ -0,0 +1,49 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Generate DUD dynamically using mkdud and xml file where variables
+# are expanded for the corresponding product/build.
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use strict;
+use warnings;
+use base "opensusebasetest";
+use testapi;
+use utils qw(zypper_call);
+use registration qw(add_suseconnect_product get_addon_fullname);
+
+sub expand_variables {
+    my ($xml) = shift;
+
+    my $file = "inst-sys/$xml";
+    my @vars = (script_output("cat $file") =~ /\{\{(\w+)\}\}/g);
```

Comment:
I would rather use `http://open.qa/api/testapi/#_get_test_data` here, to avoid risks of output being corrupted or polluted by other content.

Reply:
I'm thinking how this change would not change the whole test, because if we replace the string in memory instead of doing directly on the SUT, at the end we have to write it to the VM and that can also be a bit not safe, or is only on reading the pollution, because there is not api function for writing to the sut I think. The different with ay scenario is that we need the xml in the SUT. wdyt?

Reply:
Got it, is only this line instead of cat, good idea.

Reply:
Check again how we do it with autoyast, `get_test_data` combined with `http://open.qa/api/testapi/#_save_tmp_file` allows to provide any content to SUT with integrity being guaranteed. So SUT will just download file, without any need to modify it.

Reply:
I don't know why I didn't get along with this 'triad' of api functions before, perhaps because makes a bit dizzy unless you are clear what is relative path to what and where is happening in the worker or in the SUT, parameters are the same but the context is different each time, documentation is fine though.  Glad that you mention to me this (for 2nd or 3rd time :) ) I removed a lot of unnecessary code. For example, as a test, I was able to pass the xml modified to the chained job: https://openqa.suse.de/tests/latest?arch=x86_64&distri=sle&flavor=Online&machine=64bit&test=dud_development_tools_dev&version=15-SP3#downloads. Now I will work on passing the DUD with the right image.